iambic/config/wizard.py

```diff
@@ -1138,50 +1198,94 @@ def configuration_wizard_aws(self):
+        elif self.config.extends[0].key == ExtendsConfigKey.AWS_SECRETS_MANAGER:
+            response = AWS_SECRETS
+        elif self.config.extends[0].key == ExtendsConfigKey.LOCAL_FILE:
+            response = LOCALLY
+
+        if response == AWS_SECRETS:
+            region = set_aws_region(
+                "What region should the secret be created in?",
+                self.default_region,
+            )
 
-        if not questionary.confirm(f"{question_text}?").unsafe_ask():
-            self.config.secrets = {}
-            return
+            question_text = "Create the secret"
+            role_name = IAMBIC_SPOKE_ROLE_NAME
+            role_account_id = self.hub_account_id
 
-        if role_name and (aws_account := self.aws_account_map.get(role_account_id)):
-            session = asyncio.run(aws_account.get_boto3_session(region_name=region))
-        else:
-            session = boto3.Session(region_name=region)
+            if not questionary.confirm(f"{question_text}?").unsafe_ask():
+                self.config.secrets = {}
+                return
 
-        client = session.client(service_name="secretsmanager")
-        response = client.create_secret(
-            Name=f"iambic-config-secrets-{str(uuid.uuid4())}",
-            Description="IAMbic managed secret used to store protected config values",
-            SecretString=yaml.dump({"secrets": self.config.secrets}),
-        )
+            if role_name and (aws_account := self.aws_account_map.get(role_account_id)):
+                session = asyncio.run(aws_account.get_boto3_session(region_name=region))
+            else:
+                session = boto3.Session(region_name=region)
 
-        self.config.extends = [
-            ExtendsConfig(
-                key=ExtendsConfigKey.AWS_SECRETS_MANAGER,
-                value=response["ARN"],
-                assume_role_arn=get_spoke_role_arn(
-                    self.hub_account_id, read_only=self.spoke_role_is_read_only
-                ),
+            client = session.client(service_name="secretsmanager")
+            response = client.create_secret(
+                Name=f"iambic-config-secrets-{str(uuid.uuid4())}",
+                Description="IAMbic managed secret used to store protected config values",
+                SecretString=yaml.dump({"secrets": self.config.secrets}),
             )
-        ]
-        self.config.write()
+
+            self.config.extends = [
+                ExtendsConfig(
+                    key=ExtendsConfigKey.AWS_SECRETS_MANAGER,
+                    value=response["ARN"],
+                    assume_role_arn=get_spoke_role_arn(
+                        self.hub_account_id, read_only=self.spoke_role_is_read_only
+                    ),
+                )  # type: ignore
+            ]
+            self.config.write()
+        elif response == LOCALLY:
+            # TODO: add a check to see if the file exists or if is empty and ask if they want to overwrite it
```

Comment:
For these TODOs, did you already create a JIRA ticket? If not, would you please?

Reply:
EN-2071